infra/helper.py

```diff
@@ -86,24 +147,32 @@ def main():  # pylint: disable=too-many-branches,too-many-return-statements
     else:
       args.sanitizer = 'address'
 
+  name_or_path = getattr(args, 'project', None)
```

Comment:
Why can't we just do args.project etc? It seems very hacky that we pass a "project" to every subcommand, but this isn't a supported argument in all cases?

Reply:
We don't actually pass it to every subcommand. pull_images for example doesn't take it.

Reply:
I see, in that case better to do this in those commands instead. 

We can make a decorator to make this a bit nicer. e.g. something like:

```python
def extract_project(f):
  @functools.wraps(f)
  def wrapper(args):
    is_external = args.external
    project = Project(name_or_path, is_external, build_integration_path)
    return f(project, args)

@extract_project
def build_image(project, args):
  ...
```

Reply:
good idea, that works too!